src/docs/arch-hs.dox

```diff
@@ -4,5 +4,111 @@ The History Store in WiredTiger tracks old (all but the latest committed)
 versions of records.  By having these records in storage separate from
 the current version, they can be used to service long running transactions,
 and be evicted as necessary, without interfering with activity that uses
-the most recent committed versions.
+the most recent committed versions. With introduction of History Store, only the
+newest update to a key is written to the user table while the older updates
+for the key are written to the History Store. All user tables in the database are
+backed up by a single History Store table. History Store has no outward
+visibility to the application code and only WiredTiger's internal modules can
+perform operations on the History Store table using a predefined cursor API.
+
+@section arch_hs_table Historty Store table structure
+WiredTiger uses a standard row-store table to implement History Store internally.
+The key is a combination of:
+- \c btree ID of user table this update belongs to
+- \c record key (byte-string for row-store, record number for column-store)
+- \c start timestamp for the update
+- \c counter value
+This key format above allows us to search efficiently for a given record key and read
+timestamp combination. The last part of the key is a monotonically increasing counter
+to keep the key unique in scenarios where we have multiple transactions committing at
```

Comment:
I may not understand this bit but, "keep the key unique in scenarios where we have multiple transactions committing at" isn't it to handle the scenario where we have a single transaction committing multiple updates to a key at the same timestamp or to handle out of order timestamp scenarios? I suppose multiple transactions could also do that but it wouldn't be concurrent as they'd write conflict. 

Maybe it should be simplified to say: "in scenarios where a key has multiple updates occurring for the same commit timestamp." rather than trying to explain the explicit cases that that can happen in.

Reply:
Fixed.